src/python/pants/option/arg_splitter.py

```diff
@@ -94,7 +90,7 @@ def __init__(self, known_scope_infos: Iterable[ScopeInfo], buildroot: str) -> No
             str
         ] = []  # In reverse order, for efficient popping off the end.
         self._help_request: Optional[
-            HelpRequest
+            NativeGoalRequest
         ] = None  # Will be set if we encounter any help flags.
```

Comment:
You'd want to rename this and the other "help request" variables in this file to be "builtin goal".

Note that `arg_splitter.py` is generic, it's what handles converting `./pants test f.py` into `['./pants', 'test', 'f.py']`. So your change is meant to make it so it's no longer special casing the help system in particular. Rather, it's special casing _any_ builtin goal, whether that be `./pants help` or the upcoming `./pants garbage` you're adding.